### ISSUE TITLE
VaTextInput missing Required text

### ISSUE THREAD

**Issue by imelda-guzman** [department-of-veterans-affairs/vets-design-system-documentation#1125]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

Upgrading versions of component-library and web-components are not displaying the VaTextInput required text. I get two different behaviors for different versions.

When upgrading to `component-library 10.8.0` and `web-components 4.7.0` all my text inputs that have the required attribute look like this:

<img width="400" alt="missing-required-text" src="https://user-images.githubusercontent.com/748095/186228521-3cf90d80-82b0-4380-91d7-be402e3d1bc7.jpeg">

When I upgrade to latest versions of `component-library 11.5.0` and `web-components 4.12.0` the required label is gone even though I have set the required attribute on the text input.  I can see the span tag in the DOM.

<img width="350" alt="missing-required-labels" src="https://user-images.githubusercontent.com/748095/186229662-c7d6637e-3cd8-47e9-8265-469da4a21faa.jpeg">

<img width="318" alt="dom-required-span" src="https://user-images.githubusercontent.com/748095/186229660-46bcdaff-c7fb-45a4-bba1-cdebe449c7f1.png">

## What I expected to happen
I would expect to upgrade versions and be able to set the required attribute and see the Required text on text inputs as seen below

<img width="400" alt="correct-input-ui" src="https://user-images.githubusercontent.com/748095/186233582-edeea511-8836-47b9-9f32-081bb5c9bb5a.jpeg">

## Reproducing
We are currently in the process of updating some of our packages but even with the new package updates we get the same behavior for the VaTextInput.

**Our  OLD package.json**
```
{
  "name": "vas-config-web",
  "version": "1.0.10",
  "private": true,
  "dependencies": {
    "@department-of-veterans-affairs/component-library": "^10.0.0",
    "@department-of-veterans-affairs/formation": "^7.0.2",
    "@department-of-veterans-affairs/web-components": "^4.0.0",
    "@fortawesome/fontawesome-free": "^5.15.4",
    "@tippyjs/react": "^4.2.6",
    "axios": "^0.21.1",
    "eslint-config-react-app": "7.0.0",
    "foundation-sites": "5",
    "js-cookie": "^3.0.1",
    "jwt-decode": "^3.1.2",
    "lodash": "^4.17.21",
    "mobx": "^6.3.7",
    "mobx-react-lite": "^3.2.2",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-helmet": "^6.1.0",
    "react-idle-timer": "^4.6.4",
    "react-router-dom": "^5.2.0",
    "react-scripts": "^5.0.0",
    "react-select": "^5.3.0",
    "react-table": "^7.7.0",
    "uswds": "^1.6.10",
    "web-vitals": "^1.0.1"
  },
  "devDependencies": {
    "@axe-core/react": "^4.2.2",
    "@babel/cli": "^7.16.0",
    "@babel/core": "^7.16.0",
    "@babel/eslint-parser": "^7.16.3",
    "@babel/preset-env": "^7.16.4",
    "@babel/preset-react": "^7.16.0",
    "@testing-library/dom": "^8.11.1",
    "@testing-library/jest-dom": "^5.15.0",
    "@testing-library/react": "^12.1.2",
    "@testing-library/react-hooks": "^7.0.2",
    "@testing-library/user-event": "^13.5.0",
    "@types/jest": "^27.4.1",
    "@types/js-cookie": "^3.0.1",
    "@types/lodash": "^4.14.180",
    "@types/node": "^17.0.21",
    "@types/react": "^17.0.40",
    "@types/react-dom": "^17.0.13",
    "@types/react-helmet": "^6.1.5",
    "@types/react-router-dom": "^5.3.3",
    "@types/react-table": "^7.7.12",
    "concurrently": "^6.4.0",
    "eslint": "^8.0.1",
    "eslint-config-airbnb": "^19.0.4",
    "eslint-config-airbnb-typescript": "^16.1.4",
    "eslint-config-prettier": "^8.3.0",
    "eslint-config-standard": "^17.0.0-1",
    "eslint-plugin-import": "^2.25.3",
    "eslint-plugin-jsx-a11y": "^6.5.1",
    "eslint-plugin-node": "11.1.0",
    "eslint-plugin-promise": "6.0.0",
    "eslint-plugin-react": "^7.27.1",
    "eslint-plugin-react-hooks": "^4.3.0",
    "json-server": "^0.17.0",
    "msw": "^0.35.0",
    "node-sass": "^7.0.1",
    "postcss": "^8.3.11",
    "typescript": "^4.6.2",
    "uuid": "^7.0.3"
  }
```

**Our NEW package.json**
```
{
  "name": "vas-config-web",
  "version": "1.0.10",
  "private": true,
  "dependencies": {
    "@department-of-veterans-affairs/component-library": "^10.0.0",
    "@department-of-veterans-affairs/formation": "^7.0.2",
    "@department-of-veterans-affairs/web-components": "^4.0.0",
    "@fortawesome/fontawesome-free": "^5.15.4",
    "@tippyjs/react": "^4.2.6",
    "axios": "^0.21.1",
    "eslint-config-react-app": "7.0.0",
    "foundation-sites": "5",
    "js-cookie": "^3.0.1",
    "jwt-decode": "^3.1.2",
    "lodash": "^4.17.21",
    "mobx": "^6.6.1",
    "mobx-react-lite": "^3.4.0",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-helmet": "^6.1.0",
    "react-idle-timer": "^4.6.4",
    "react-router-dom": "^5.3.3",
    "react-scripts": "^5.0.0",
    "react-select": "^5.4.0",
    "react-table": "^7.8.0",
    "uswds": "^1.6.10",
    "web-vitals": "^1.1.2"
  },
  "devDependencies": {
    "@axe-core/react": "^4.4.3",
    "@babel/cli": "^7.18.10",
    "@babel/core": "^7.18.13",
    "@babel/eslint-parser": "^7.18.9",
    "@babel/preset-env": "^7.18.10",
    "@babel/preset-react": "^7.18.6",
    "@testing-library/dom": "^8.17.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^12.1.5",
    "@testing-library/react-hooks": "^7.0.2",
    "@testing-library/user-event": "^13.5.0",
    "@types/jest": "^27.5.2",
    "@types/js-cookie": "^3.0.2",
    "@types/lodash": "^4.14.184",
    "@types/node": "^17.0.45",
    "@types/react": "^17.0.48",
    "@types/react-dom": "^17.0.17",
    "@types/react-helmet": "^6.1.5",
    "@types/react-router-dom": "^5.3.3",
    "@types/react-table": "^7.7.12",
    "concurrently": "^7.3.0",
    "eslint": "^8.22.0",
    "eslint-config-airbnb": "^19.0.4",
    "eslint-config-airbnb-typescript": "^16.2.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-config-standard": "^17.0.0",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-jsx-a11y": "^6.6.1",
    "eslint-plugin-node": "11.1.0",
    "eslint-plugin-promise": "6.0.0",
    "eslint-plugin-react": "^7.30.1",
    "eslint-plugin-react-hooks": "^4.6.0",
    "json-server": "^0.17.0",
    "msw": "^0.35.0",
    "node-sass": "^7.0.1",
    "postcss": "^8.4.16",
    "typescript": "^4.6.4",
    "uuid": "^7.0.3"
  }
```
- Device: HP Elitebook
- Browser: Microsoft Edge

Steps to reproduce:

1. Upgrade component-library to 10.8.0 or 11.5.0 and web-components to 4.7.0 or 4.12.0
2. Add VaTextInput to page and set `required` attribute.

**Comment by bkjohnson** [comment]
How are you loading and defining the web components on your site?

**Comment by imelda-guzman** [comment]
@bkjohnson we are importing all web components like this:
```
import {
  VaSelect,
  VaTextInput,
  VaAccordion,
  VaAccordionItem,
  VaCheckbox,
  VaCheckboxGroup,
  VaRadio,
  VaRadioOption
} from '@department-of-veterans-affairs/component-library/dist/react-bindings';
```
and the component

```
<VaTextInput
    data-testid="clinicName"
    required
    error={clinicFormStore.errors.name}
    label="Clinic Name"
    name="name"
    value={formData.name}
    onInput={debouncedInputHandler}
    onBlur={(
       event: React.FocusEvent<HTMLVaTextInputElement>
     ) => trimSpaceHandler(event, 'name')}
/>
```

**Comment by bkjohnson** [comment]
In [version 10.1.0](https://github.com/department-of-veterans-affairs/component-library/releases/tag/v10.1.0) we began providing built-in translations to some components. We're doing the necessary setup for that in the index file of the `component-library` package. If you follow [our guidance here](https://design.va.gov/about/developers#load-the-web-component-library) and import and call `defineCustomElements`, calling the function will let the browser know about the web components and simply making the import will cause the necessary language setup to happen.

**Comment by imelda-guzman** [comment]
@bkjohnson thanks for pointing me in the right direction.  I needed to update the import in our `index.tsx` file. Our app was still importing the old reference

```
import {
  applyPolyfills,
  defineCustomElements,
} from '@department-of-veterans-affairs/web-components/loader';
```
The Required label works in both versions `10.8.0` and `11.5.0` after I updated the correct import as detailed [here](https://design.va.gov/about/developers#load-the-web-component-library)

Thanks again for your help.
